Fix: Task Date #13

x4th opened the pull request:
# Description

Added date field for tasks.

Fixes #13

## Type of change

This is a:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (fixes or update of an existing feature)

## Checklist:

By making this contribution I ensure that:

- [x] I have read the [contribution guide](https://github.com/DevTony101/js-todo-app-indexed_db/blob/master/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [README file](https://github.com/DevTony101/js-todo-app-indexed_db#features) (if this PR introduces a new feature)
- [x] My changes generate no new warnings

DevTony101 reviewed (changes requested):
Hi there! First of all, thank you so much for your contribution!

Everything seems to work just fine but there are a few things that are bugging me:

- The date is supposed to be a "deadline" for the task, but the UI does not tell the user that, neither in the input field nor in the task header
- There is no way the user can tell if the date is optional beforehand, so the first impression would be appending a date even if it isn't necessary

What would I suggest to fix that:

- Creating a tooltip that displays what the input is intended for (either attached to the input itself or to an "info" icon right next to it)
- Put the date at the end of the message header (in the task panel, when it's created) and explicitly say what it represents (ex: "Deadline: ", "Date set: ", etc)

And finally, in regard to the PR itself:

- In order to maintain the integrity of the project it is imperative that you check all the elements in the PR checklist to ensure the quality of your contribution
- As this is a brand new feature of the app, it **must** appear on the features section of the README file (as explained in the [contributing guide](https://github.com/DevTony101/js-todo-app-indexed_db/blob/master/CONTRIBUTING.md))

I hope this does not overwhelm you, it's done with the best of the intentions. I'm looking forward to review your new changes! 👍🏽

x4th commented:
I think it's better to just state "Deadline" in the field label as that's what it essentially is, not sure why I didn't do this originally. I've renamed all occurrences of "date" to "deadline", marked the title as a required field which solves your second bullet point and moved the deadline date to be on the right side of the message header with the text in front "Deadline:".

DevTony101 reviewed (approved):
Nicely done! Everything works as expected. Thanks again for your contribution and for showing interest towards this project. 👍🏽
